client/js/lounge.js

```diff
@@ -428,16 +490,20 @@ $(function() {
 		}
 
 		if (prevMsgTime.toDateString() !== msgTime.toDateString()) {
-			prevMsg.append(render("date-marker", {msgDate: msgTime}));
+			var parent = prevMsg.parent();
+			if (parent.hasClass("condensed")) {
+				prevMsg = parent;
+			}
+			prevMsg.after(render("date-marker", {msgDate: msgTime}));
```

Comment:
In that case shouldn't it be `after(prevMsg)`? I'm just not seeing this ends up in the DOM.